lib/dato/uploads.rb

```diff
@@ -0,0 +1,108 @@
+module Dato
+  class Uploads
+    BASE_ITEM_URL = 'https://site-api.datocms.com/upload-requests'
+
+    def initialize(api_token)
+      @auth_header = "Bearer #{api_token}"
+
+      headers = {
+        'Authorization' => @auth_header,
+        'Accept' => 'application/json',
+        'X-Api-Version' => 3
+      }
+      @http_headers = HTTP.headers(headers)
+    end
+
+    def create_from_url(url:, attributes: {}, meta: {})
+      file = download_file_from_url(url)
+      create(path_to_file: file.path, attributes:, meta:)
+    end
+
+    def create(path_to_file:, attributes:, meta: nil)
+      # Step 1: Request file upload permission
+      request = request_file_upload_permission(path_to_file)
+      upload_url = request[:url]
+      upload_id = request[:id]
+
+      # Step 2: Upload file to storage bucket
+      upload_file_to_bucket(url: upload_url, path: path_to_file)
+
+      # Step 3: Create the actual upload
+      upload_file_to_dato(upload_id:, attributes:, meta:)
+
+      # Step 4: Retrieve Job result (via polling...)
```

Comment:
make sure to return the id of the image

Reply:
`{ upload_id: '67094498' }`

Reply:
Makes more sense, than the JobID it currenty returns... 
Should it return the `upload_id` or a hash with the `upload_id`?

Reply:
the best would be a hash `{ upload_id: '67094498' }` to be similar to the JS Dato version